app/helpers/layout_helper.rb

```diff
@@ -347,6 +347,18 @@ def fullscreen_button(element = "$(this).prev()")
   private
 
   def authorized_associations(associations)
-    associations.included_modules.include?(Authorizable) ? associations.authorized : associations
+    if associations.included_modules.include?(Authorizable)
+      if associations.respond_to?(:klass)
+        associations.authorized(authorized_assocations_permission_name(associations.klass), associations.klass)
+      else
+        associations.authorized(authorized_assocations_permission_name(associations), associations)
+      end
+    else
+      associations
+    end
+  end
+
+  def authorized_assocations_permission_name(klass)
+    "view_#{klass.to_s.downcase.pluralize}"
```

Comment:
Good catch with the `downcase`.

> I also wonder if there are plugins that add stuff to the taxonomy form and that might have more complex permission name rules?

They could wrap the method, that should be fine as long as they can decide upon a given `klass`.

> We might need a test to loop around taxonomy-related things and check this resolves to a known permission name too?

What do you want me to test? Maybe that `authorized_associations` method can handle situation when there's no permission for associated the resource? What would be the right behaviour in this case? Currently, `authorized` scope wouldn't limit on specific permission but take resources that are granted via any permission (like before this patch). Maybe such resource shouldn't include `Authorizable` module in the first place...